### DISCUSSION TITLE
Microsoft Word/Google Docs-quality comments

### DISCUSSION THREAD

**Discussion by kendonB** [quarto-dev/quarto-cli#3200] [Feature Requests]
It would be great to be able to comment on a document in a way that works as well as in Microsoft Word / Google Docs. I'm sure this will be very hard, but it is one of the main points in favour of using one of these two systems over Quarto or LaTeX.

The simplest version of these comments could be defined by:

- A location in the text
- An author 

Then it just displays as a bubble on the side.

So, potentially an interface in the source could be something like:
```
<!--
# Kenny Bell (Comment heading text)
Normal text
-->
```
For Word output, they could be converted to Word comments; for LaTeX output, they could become `todonotes`.

Once a simple version is working, they could then get other features like:
 - An end location in the text, visualised by highlighted text pointing to the comment
 - Replies

**Comment by rgaiacs** [comment]
Hello @kendonB and thanks for the message.

In a Quarto to HTML workflow, you can use [Hypothesis](https://web.hypothes.is/) to have people commenting to your document as they would with Microsoft Word/Google Docs. Hypothesis will do their best to keep track to update the comments based in the changes of your document.

In a Quarto to LaTeX workflow, you can use the mentioned [todonotes](https://ctan.org/pkg/todonotes) package as Pandoc is preserve LaTeX code. For example,

```
Some text \todo{Make a cake \ldots} more text.
```

will be converted by `pandoc --from markdown --to latex` to

``` 
Some text \todo{Make a cake \ldots} more text.
```
In a Quarto to Microsoft Word workflow, you can use a filter (see https://github.com/jgm/pandoc/issues/2994).

**Comment by kendonB** [comment]
I recall hearing that Quarto -> MS Word was going to be an allowable workflow - does that work for comments?

**Reply by rgaiacs** [reply]
I fixed a typo in my previous reply. In Quarto -> MS Word, you can use a filter. I don't know if the team behind Quarto is already working to include a similar filter in the Quarto release.
